lib/src/realm_object.dart

```diff
@@ -364,12 +397,18 @@ mixin RealmObject on RealmEntity implements Finalizable {
   late final DynamicRealmObject dynamic = DynamicRealmObject._(this);
 
   /// Creates a frozen snapshot of this [RealmObject].
-  RealmObject freeze() => freezeObject(this);
+  RealmObjectBase freeze() => freezeObject(this);
 }
 
+/// @nodoc
+mixin RealmObject on RealmObjectBase {}
```

Comment:
Because we want to restrict some API on `RealmObject` - i.e. prevent embedded objects from using that API. For example, `Realm.add<T extends RealmObject>` should only accept top level objects and not embedded objects.

Reply:
We decided to leave as is